lib/jet_plugin_sdk/tenant_man/tenants/tenant.ex

```diff
@@ -92,57 +111,51 @@ defmodule JetPluginSDK.TenantMan.Tenants.Tenant do
         }
       end
 
-      defoverridable child_spec: 1
+      @impl unquote(__MODULE__)
+      def terminate(_reason, _state) do
+        :ok
+      end
 
-      @behaviour JetPluginSDK.TenantMan.Tenants.Tenant
+      defoverridable child_spec: 1, terminate: 2
     end
   end
 
-  @enforce_keys [:tenant_id, :tenant_module, :tenant]
+  @enforce_keys [:tenant_module, :tenant]
   defstruct [
-    :tenant_id,
     :tenant_module,
     :tenant,
     :tenant_state
   ]
 
   @type t() :: %__MODULE__{
-          tenant_id: tenant_id(),
           tenant_module: module(),
           tenant: tenant_schema(),
           tenant_state: state()
         }
 
-  @type name() :: {module(), tenant_id()}
-
   @type start_link_opts() :: [
-          tenant_id: tenant_id(),
+          name: Registry.name(),
           tenant_module: module(),
-          tenant: tenant_schema(),
-          name: name()
+          tenant: tenant_schema()
         ]
 
-  @spec name(tenant_module :: module(), tenant_id()) :: name()
-  def name(tenant_module, tenant_id) do
-    {tenant_module, tenant_id}
-  end
-
-  @spec fetch_tenant(tenant_module :: module(), tenant_id()) :: {:ok, tenant_schema()} | :error
+  @spec fetch_tenant(tenant_module :: module(), tenant_id :: tenant_id()) ::
+          {:ok, tenant_schema()} | :error
   def fetch_tenant(tenant_module, tenant_id) do
-    with(
-      {:ok, pid} <-
-        JetPluginSDK.TenantMan.Tenants.Supervisor.whereis_tenant(tenant_id, tenant_module)
-    ) do
+    with {:ok, pid} <- Registry.whereis(tenant_module, tenant_id) do
       GenServer.call(pid, {:"$tenant_man", :fetch_tenant})
     end
   end
 
-  @spec update_config(tenant_module :: module(), tenant_id(), new_config :: tenant_config()) ::
-          term()
-  def update_config(tenant_module, tenant_id, new_config) do
-    case JetPluginSDK.TenantMan.Tenants.Supervisor.whereis_tenant(tenant_id, tenant_module) do
+  @spec update_config(
+          tenant_module :: module(),
+          tenant_id :: tenant_id(),
+          config :: tenant_config()
+        ) :: term()
+  def update_config(tenant_module, tenant_id, config) do
```

Comment:
callback 现在改成了 handle_update，这里也改成 update 吧，另外为下一步做 capabilities 的 update 准备一下

```suggestion
  @spec update(
          tenant_module :: module(),
          tenant_id :: tenant_id(),
          params :: %{config: tenant_config()}
        ) :: term()
  def update(tenant_module, tenant_id, params) do
```